src/visitable.cpp

```diff
@@ -72,14 +72,24 @@ bool visitable<T>::has_item_with( const std::function<bool( const item & )> &fil
     } ) == VisitResponse::ABORT;
 }
 
+// Sums the two terms, being careful to not trigger any overflows.
+// Doesn't handle underflow.
+static int sum_no_wrap( int a, int b ) {
```

Comment:
That looks good.

It could be worth putting it in `cata_utility.h` as a template function as it has broader applications throughout the codebase